src/screens/crewStationScreen.cpp

```diff
@@ -144,6 +147,34 @@ void CrewStationScreen::update(float delta)
                 break;
             }
         }
+
+#ifndef __ANDROID__
+        // If engine sounds are enabled and we have impulse, loop the engine sound.
```

Comment:
Not so happy that this is duplicated code with the main screen.

Reply:
Me neither, but the music code is already duplicated between ScreenMainScreen and CrewStationScreen.

I'm having trouble figuring out the best place to put both of them — in a new background channel managed by SF's SoundManager, or in a ScreenComponent in EE, or by rewriting the Main Screen to inherit from CrewStationScreen (or creating a more generic screen class that both ScreenMainScreen and CrewStationScreen)

Reply:
I think a new class that just manages the engine sounds would be a good way. And then just create this new class from both different screens.

Reply:
Moved impulse sound to its own class under ScreenComponents in 85cd2ab.